### ISSUE TITLE
Failure to Reproduce Results of RoBERTa

### ISSUE THREAD

**Issue by KexuanShi** [nathanielyvo/WUDI-Merging#6]
Following the current implementation of WUDI in the wudi_main.py, we can't reproduce the GLUE Benchmark Results with RoBERTa-base. Specifically, when I set the iter_num=300, the absolute average performance is 0.5243；If the iter_num=600, the absolute average performance is 0.5828.  The results deviate significantly from those reported in the paper.

**Comment by nathanielyvo** [comment]
For the experiments on roberta-base, we used the codebase and checkpoints in: https://github.com/LZY-the-boys/Twin-Merging. You can add our method on it to evaluate the results.

Please note that the results we report are normalized accuracy, not absolute accuracy, so please take this into account when making comparisons. 

If you have any further questions, feel free to reach out at any time.

**Comment by KexuanShi** [comment]
I have carefully examined the absolute average accuracy in:  https://github.com/LZY-the-boys/Twin-Merging, which is around 0.8256. This is very close to the performance of my fine-tuned checkpoints. According to reported values in the paper, the performance of WUDI should be close to 0.70 for the absolute performance. Therefore, I am concerned that my deployment of the WUDI method may have overlooked certain implementation details. Nevertheless, the deployment successfully reproduces the reported results in the vision experiments.

**Comment by nathanielyvo** [comment]
We've uploaded a draft version of roberta. You can follow the twin-merging  to install the environment and download  the checkpoints.

then, You can run our method by:

```
source scripts_base.sh
run_ta_decompose
```
When the learning rate is set to 2e-5, you can obtain the results reported in our paper. When the learning rate is 1e-5, the normalized accuracy is 84.2.

If you have any further questions, please feel free to contact us.

**Comment by KexuanShi** [comment]
<img width="1151" height="626" alt="Image" src="https://github.com/user-attachments/assets/abe04606-e70a-4f0b-a5a1-4285ff3514bd" />

As shown in the above graph, the core code is the part that's commented out, right?

**Comment by KexuanShi** [comment]
<img width="2225" height="710" alt="Image" src="https://github.com/user-attachments/assets/57094eaf-0c78-496c-9b51-ceee44b30ae3" />

Following your released draft code, the results have been presented in the above graph. Unfortunately, the performance is still far away from the reported value in the paper. Please note that the finetuned checkpoints we used were obtained from TWIN-Merging and DARE. The avg performance is even better than yours.

**Comment by nathanielyvo** [comment]
<img width="1096" height="221" alt="Image" src="https://github.com/user-attachments/assets/98a9fcb9-24e6-49c0-a441-6c3c16568206" />

Sorry, I didn’t quite understand what you meant by “avg performance is even better than yours” — could you clarify which checkpoints you were referring to? We redeployed the code from the GitHub repository and obtained the same results as shown in the image above. Could you confirm whether you used exactly the same checkpoints and followed the same environment setup process as described in the Twin-Merging implementation?

ps: the checkpoints we and twin-merging used is in: https://huggingface.co/lu-vae/roberta-glue

**Comment by nathanielyvo** [comment]
> <img alt="Image" width="1151" height="626" src="https://private-user-images.githubusercontent.com/88844484/473958515-abe04606-e70a-4f0b-a5a1-4285ff3514bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMTc3OTYsIm5iZiI6MTc1NDMxNzQ5NiwicGF0aCI6Ii84ODg0NDQ4NC80NzM5NTg1MTUtYWJlMDQ2MDYtZTcwYS00ZjBiLWE1YTEtNDI4NWZmMzUxNGJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA0VDE0MjQ1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiZmU0ZDBmMjg2MTY4OWU1NmJlMTU2YmNiZmY1MTcxYTNiMjVhNGI0NzkwZGU2ZjY0YmMxNDhhY2QxNTUwMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tb1MNLMp7g2Voj0lrFyKkDTSA6G_wYjAeGUNpnrKEMM">
> As shown in the above graph, the core code is the part that's commented out, right?

The core code part is in `def run_awd_merge` in `run_merge.py`，`def get_merge_vector` in `lamassu.py` and `def task_arithmetic_decompose` in `merge.py`. You can refer to the code here to try reproducing the results on your own.

**Comment by nathanielyvo** [comment]
> <img alt="Image" width="2000" height="710" src="https://private-user-images.githubusercontent.com/88844484/474050352-57094eaf-0c78-496c-9b51-ceee44b30ae3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0NTg1MjAsIm5iZiI6MTc1NDQ1ODIyMCwicGF0aCI6Ii84ODg0NDQ4NC80NzQwNTAzNTItNTcwOTRlYWYtMGM3OC00OTZjLTliNTEtY2VlZTQ0YjMwYWUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDA1MzAyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MmE3ZTZmNjE0OGVmZmZhOTkwMzEzNTUzODFjZjEwNGY4ZDRjNjhhZDc1ZDk5NWEzNGQ4NGNiYTk0NWI0YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ahPa4UO-fHDUNKpv53y3FOr7Z-nNpDPfzKlPgwLFnq8">
> Following your released draft code, the results have been presented in the above graph. Unfortunately, the performance is still far away from the reported value in the paper. Please note that the finetuned checkpoints we used were obtained from TWIN-Merging and DARE. The avg performance is even better than yours.

Hi Kexuan,
I hope everything is going well.
Just wanted to kindly check in — have you been able to successfully reproduce it?
If you ran into any issues or would like any further assistance, please don’t hesitate to let me know. I’d be happy to help.
